gemfirexd/client/src/main/java/io/snappydata/jdbc/ClientPoolDriver.java

```diff
@@ -19,123 +19,151 @@
-    }
+  private static String SNAPPY_PROTOCOL = "jdbc:snappydata:";
+
+  private final static String SUBPROTOCOL = "(pool:){1}";
 
-    @Override
-    protected boolean useThriftProtocol(Matcher m) {
-        String subProtocolGroup = m.group(2);
-        String protocol = m.group(1);
-        // if USE_THRIFT_AS_DEFAULT_PROP has been set explicitly then use that
-        // for default value else use the protocol string (jdbc:snappydata://
-        //   defaults to thrift while jdbc:gemfirexd:// defaults to old DRDA)
-        return subProtocolGroup == null || subProtocolGroup.length() == 0
-                ? ClientSharedUtils.isUsingThrift(
-                protocol.equalsIgnoreCase(SNAPPY_PROTOCOL))
-                : "thrift:pool:".equalsIgnoreCase(subProtocolGroup)
-                || "pool:".equalsIgnoreCase(subProtocolGroup);
+  protected final static String URL_PREFIX_REGEX = "(" + SNAPPY_PROTOCOL + ")";
+  protected final static String URL_SUFFIX_REGEX =
+      "//(([^:]+:[0-9]+)|([^\\[]+\\[[0-9]+\\]))(/(snappydata;)?;?(.*)?)?";
+
+  private final static Pattern PROTOCOL_PATTERN = Pattern.compile(URL_PREFIX_REGEX +
+      SUBPROTOCOL + "//.*", Pattern.CASE_INSENSITIVE);
+
+  private final static Pattern URL_PATTERN = Pattern.compile(URL_PREFIX_REGEX +
+      SUBPROTOCOL + URL_SUFFIX_REGEX, Pattern.CASE_INSENSITIVE);
+
+  private static SQLException exceptionsOnLoadDriver__ = null;
+
+  static {
+    try {
+      ClientPoolDriver driver = new ClientPoolDriver();
+      java.sql.DriverManager.registerDriver(driver);
+
+    } catch (SQLException e) {
+      // A null log writer is passed, because jdbc 1 sql exceptions are
+      // automatically traced
+      exceptionsOnLoadDriver__ = ClientExceptionUtil.newSQLException(
+          SQLState.JDBC_DRIVER_REGISTER, e);
+    }
+    // This may possibly hit the race-condition bug of java 1.1.
+    // The Configuration static clause should execute before the following line
+    // does.
+    if (ClientConfiguration.exceptionsOnLoadResources != null) {
+      exceptionsOnLoadDriver__ = Utils.accumulateSQLException(
+          ClientConfiguration.exceptionsOnLoadResources,
+          exceptionsOnLoadDriver__);
     }
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public boolean acceptsURL(String url) throws SQLException {
+    return (url != null && URL_PATTERN.matcher(url).matches());
+  }
 
-    @Override
-    protected Matcher matchURL(String url) {
-        return URL_PATTERN.matcher(url);
+  @Override
+  public Connection connect(String url, Properties properties) throws SQLException {
+
+    if (!acceptsURL(url)) {
+      return null;
     }
 
-    @Override
-    protected Matcher matchProtocol(String url) {
-        return PROTOCOL_PATTERN.matcher(url);
+    properties = (properties == null) ? new Properties() : properties;
+    String clientDriverURL = url.replaceFirst(PROTOCOL_PATTERN.matcher(url)
+        .replaceFirst(SUBPROTOCOL), "");
```

Comment:
Can't this be simplified to use a single replace rather than two replaceFirst calls? For example have `PROTOCOL_PATTERN = Pattern.compile(URL_PREFIX_REGEX + SUBPROTOCOL, Pattern.CASE_INSENSITIVE)`, then PROTOCOL_PATTERN.matcher(url).replaceFirst(SNAPPY_PROTOCOL)

Reply:
simplified and tested, also added the URL case sensitive  scala test.